File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        
        res=""
        cnt=0
        
        for char in s:
            if char=='(':
                if cnt>0:
                    res+=char
                cnt+=1
                
            else:
                
                if cnt>1:
                    res+=char
                 
                cnt-=1
                
                
                
        return res        
               
                
               
            
           
                
                
             

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        
        int cnt=0;
        string ans;
        for(auto it:s)
        {
            
            if(it=='(')
            {
                if(cnt>0)
                {
                    ans+=it;
                }
                cnt++;
            }
            else{
                
                if(cnt>1)
                {
                    ans+=it;
                    
                }
                cnt--;
            }
        }
        return ans;
    }
};